packages/core-mobile/app/seedless/services/SeedlessSession.ts

```diff
@@ -121,101 +116,68 @@ class SeedlessSessionManager {
       mfaReceipt
     )
 
-    const sessionResponse = signResponse.requiresMfa()
-      ? signResponse.mfaSessionInfo()
-      : signResponse.data()
-
-    if (sessionResponse) {
-      await SignerSessionManager.createFromSessionInfo(
-        envInterface,
-        SEEDLESS_ORG_ID,
-        sessionResponse,
-        this.sessionStorage
-      )
+    if (response.requiresMfa()) {
+      // if MFA is required, we cannot access session data until MFA verification is complete
+      // thus, we are just storing the MFA client here (for verification operations)
+      const client = await response.mfaClient()
+      if (!client) throw new Error('MFA client is missing')
+      this.signerClient = client
+    } else {
+      // when MFA is not required or MFA verification is complete
+      // we can safely access session data
+      const sessionData = response.data()
+
+      // destroy outdated client instance
+      this.signerClient = null
```

Comment:
here we simply "destroy" the current client. next time the app needs a client, it will construct it based on the latest session data that we just save here.